doc/rst/language/spec/procedures.rst

```diff
@@ -532,114 +630,142 @@ model.
-  * unions ``const ref``
-  * synchronization types (``atomic``, ``sync``)
+  In the current compiler implementation, ``const`` means ``const in``
+  for the following types:
+
+     * for ``bool``, ``int``, ``uint``, ``real``, and ``imag`` with
+       widths less than or equal to 64 bits
+
+     * for ``complex`` with widths less than or equal to  128 bits
+
+     * for ``borrowed`` and ``unmanaged`` class types
+
+     * for values of enumerated type
 
-The ``const`` intent for tuples applies the ``const`` intent to each tuple
-component as if it was passed as a separate argument.
-See :ref:`Tuple_Argument_Intents`.
 
 .. _The_Default_Intent:
 
 The Default Intent
-^^^^^^^^^^^^^^^^^^
+~~~~~~~~~~~~~~~~~~
 
-When no intent is specified for a formal argument, the *default
-intent* is applied. It is designed to take the most natural/least
-surprising action for the argument, based on its type. In practice,
-this is ``const`` for most types (as defined by
-:ref:`The_Const_Intent`) to avoid surprises for programmers coming from
-languages where everything is passed by ``in`` or ``ref`` intent by default.
+When no intent is specified for a formal argument, the *default intent*
+is applied. The default intent is designed to take the most natural/least
+surprising action for the argument.
 
-Exceptions are made for types where modification is considered part of their nature.
-In particular, the the default intent for the following types is ``ref``:
+For most types, the default intent is the same as the ``const`` intent.
+See the description of the ``const`` intent in :ref:`The_Const_intent`
+for details including  assumptions that the compiler can make about
+default intent and ``const`` formals when optimizing.
 
-  * ``atomic``
-  * ``sync``
+However, the default intent for ``atomic`` and ``sync`` is ``ref`` because
+modification is considered part of their nature.
 
-Default argument passing for tuples applies the default
-argument passing strategy to each tuple component as if it
-was passed as a separate argument. See :ref:`Tuple_Argument_Intents`.
+Additionally, tuples have a special default argument intent. See
+:ref:`Tuple_Argument_Intents` for details.
 
-.. _Default_Intent_for_owned_and_shared:
+.. _Intents_for_owned_and_shared:
 
-Default Intent for ’owned’ and ’shared’
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Default and 'const' Intents for ’owned’ and ’shared’
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The default intent for :type:`~OwnedObject.owned` and
 :type:`~SharedObject.shared` arguments is ``const``. To transfer the ownership
 from an :type:`~OwnedObject.owned` actual argument or to share the ownership
 with a :type:`~SharedObject.shared` actual argument, the formal argument can use
-the ``in`` or ``const in`` intent.
+the ``in`` or ``const in`` intent. Note that passing an actual argument of type
+:type:`~OwnedObject.owned` to a ``const in`` formal can change the
```

Comment:
Line 762: "can use" --> "needs to use"
Line 763: Also add `inout` intent?
Line 764: "can change" --> "will change"

Reply:
>  Line 763: Also add inout intent?

I wouldn't describe inout as transferring ownership because it has to use assignment because the copy cannot be elided.